app/src/main/java/protect/card_locker/MainActivity.kt

```diff
@@ -0,0 +1,924 @@
+    private lateinit var binding: MainActivityBinding
+    private lateinit var contentMainBinding: ContentMainBinding
+    private lateinit var mDatabase: SQLiteDatabase
+    private lateinit var mAdapter: LoyaltyCardCursorAdapter
+    private var mCurrentActionMode: ActionMode? = null
+    private var mSearchView: SearchView? = null
+    private var mLoyaltyCardCount = 0
+    @JvmField
+    var mFilter: String = ""
+    private var currentQuery = ""
+    private var finalQuery = ""
+    protected var mGroup: Any? = null
+    protected var mOrder: LoyaltyCardOrder = LoyaltyCardOrder.Alpha
+    protected var mOrderDirection: LoyaltyCardOrderDirection = LoyaltyCardOrderDirection.Ascending
+    protected var selectedTab: Int = 0
+    private lateinit var groupsTabLayout: TabLayout
+    private lateinit var mUpdateLoyaltyCardListRunnable: Runnable
+    private lateinit var mBarcodeScannerLauncher: ActivityResultLauncher<Intent?>
+    private lateinit var mSettingsLauncher: ActivityResultLauncher<Intent?>
+
+    private val mCurrentActionModeCallback: ActionMode.Callback = object : ActionMode.Callback {
+        override fun onCreateActionMode(inputMode: ActionMode, inputMenu: Menu?): Boolean {
+            inputMode.getMenuInflater().inflate(R.menu.card_longclick_menu, inputMenu)
+            return true
+        }
+
+        override fun onPrepareActionMode(inputMode: ActionMode?, inputMenu: Menu?): Boolean {
+            return false
+        }
+
+        override fun onActionItemClicked(inputMode: ActionMode, inputItem: MenuItem): Boolean {
+            when (inputItem.getItemId()) {
+                R.id.action_share -> {
+                    val importURIHelper = ImportURIHelper(this@MainActivity)
+                    try {
+                        importURIHelper.startShareIntent(mAdapter.getSelectedItems())
+                    } catch (e: UnsupportedEncodingException) {
+                        Toast.makeText(
+                            this@MainActivity,
+                            R.string.failedGeneratingShareURL,
+                            Toast.LENGTH_LONG
+                        ).show()
+                        e.printStackTrace()
+                    }
+                    inputMode.finish()
+                    return true
+                }
+                R.id.action_edit -> {
+                    require(mAdapter.getSelectedItemCount() == 1) { "Cannot edit more than 1 card at a time" }
+
+                    val intent = Intent(getApplicationContext(), LoyaltyCardEditActivity::class.java)
+                    val bundle = Bundle()
+                    bundle.putInt(
+                        LoyaltyCardEditActivity.BUNDLE_ID,
+                        mAdapter.getSelectedItems().get(0).id
+                    )
+                    bundle.putBoolean(LoyaltyCardEditActivity.BUNDLE_UPDATE, true)
+                    intent.putExtras(bundle)
+                    startActivity(intent)
```

Comment:
I think we can reduce the amount of temporary variables if we use apply

```suggestion
                    startActivity(
                        Intent(getApplicationContext(), LoyaltyCardEditActivity::class.java).apply {
                            putExtras(
                                Bundle().apply {
                                    putInt(
                                        LoyaltyCardEditActivity.BUNDLE_ID,
                                        mAdapter.getSelectedItems().get(0).id
                                    )
                                    putBoolean(LoyaltyCardEditActivity.BUNDLE_UPDATE, true)
                                )
                            }
                        )
```